experiment/invitations-accepter/main.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/sirupsen/logrus"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+
+	"k8s.io/test-infra/prow/config/secret"
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/github"
+)
+
+type options struct {
+	github flagutil.GitHubOptions
+
+	dryRun bool
+}
+
+type githubClient interface {
+	ListCurrentUserRepoInvitations() ([]github.UserRepoInvitation, error)
+	AcceptUserRepoInvitation(invitationID int) error
+	BotUser() (*github.UserData, error)
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	fs.BoolVar(&o.dryRun, "dry-run", true, "Dry run for testing. Uses API tokens but does not mutate.")
+
+	o.github.AddFlags(fs)
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Fatal("could not parse input")
+	}
+	return o
+}
+
+func (o *options) Validate() error {
+	if err := o.github.Validate(o.dryRun); err != nil {
+		return err
+	}
+	return nil
+}
+
+func main() {
+	o := gatherOptions()
+
+	sa := &secret.Agent{}
+	if err := sa.Start([]string{o.github.TokenPath}); err != nil {
+		logrus.WithError(err).Fatal("Error starting secrets agent.")
+	}
+
+	gc, err := o.github.GitHubClient(sa, o.dryRun)
+	if err != nil {
+		logrus.WithError(err).Fatal("Error getting GitHub client.")
+	}
+	gc.Throttle(300, 300)
+
+	if err := acceptInvitations(gc, o.dryRun); err != nil {
+		logrus.WithError(err).Fatal("errors occurred.")
+	}
+}
+
+func acceptInvitations(gc githubClient, dryRun bool) error {
+	var errs []error
+
+	botUser, err := gc.BotUser()
+	if err != nil {
+		return fmt.Errorf("couldn't get bot's user name: %v", err)
+	}
+	logger := logrus.WithField("bot-user", botUser.Login)
+
+	repoInvitations, err := gc.ListCurrentUserRepoInvitations()
+	if err != nil {
+		logger.WithError(err).Fatal("couldn't get repo invitations for the authenticated user")
+	}
+
+	if len(repoInvitations) == 0 {
+		logger.Info("No pending invitations have been found.")
+		return nil
+	}
+
+	for _, inv := range repoInvitations {
+		logger = logger.WithFields(logrus.Fields{"invitation-id": inv.InvitationID, "repo": inv.Repository.FullName})
```

Comment:
```suggestion
		logger := logger.WithFields(logrus.Fields{"invitation-id": inv.InvitationID, "repo": inv.Repository.FullName})
```

We don't want to override the logger var, if someone logs something after the `for` loop later it would have the fields from here